packages/notifications/push/amplify_push_notifications/lib/amplify_push_notifications.dart

```diff
@@ -132,16 +132,16 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
       return;
     }
 
-    // Block configure if registering device is not complete
-    // final deviceToken = await onTokenReceived.first;
-    // await _registerDevice(deviceToken);
-
     // Initialize Endpoint Client
     await _serviceProviderClient.init(
       config: config,
       authProviderRepo: authProviderRepo,
     );
 
+    // Block configure if registering device is not complete
+    final deviceToken = await onTokenReceived.first;
+    await _registerDevice(deviceToken);
```

Comment:
Not sure about the blocking.

For example, if the device is offline when a end user launches the App, where `registerForRemoteNotifications` cannot go through and get a token back, the plugin will just hang at this point? I would expect something like this, it also handles the case when the deviceToken changed. What do you think?

```dart
onTokenReceived.listen((deviceToken) {
  _registerDevice(deviceToken);
});
```

Reply:
If it's offline - it should fail IIRC so that can be logged and tried again later. But we should try to await it at least once in case it fails for other reasons.

Reply:
OK, in that case, we'd need to send a `FlutterError`  to `onTokenReceived` event channel on `didFailToRegisterForRemoteNotificationsWithError` (iOS for example) , so the first `await onTokenReceived.first` can actually fail.

Reply:
Yes we should definitely be doing that

Reply:
Did some testing on iOS. On a fresh App installation, when call `registerForRemoteNotifications` when the device is offline, the system can still return a device token, and no error was delegated to `didFailToRegisterForRemoteNotificationsWithError`. Not sure what's the behavior on Android.

Did more digging on `didFailToRegisterForRemoteNotificationsWithError`, this delegate method called only when the App is incorrectly signed. My assumption that it would caught error when device is offline is invalid 😅

But yes it's still necessary to send an error to `onTokenReceived`.